docs/shortfin/llm/user/e2e_llama8b_mi300x.md

```diff
@@ -0,0 +1,230 @@
+# LLama 8b GPU Instructions on MI300X
+
+**NOTE: This was ran on the `mi300x-3` system**
+
+## Setup
+
+We will use an example with `llama_8b_f16_decomposed` in order to describe the process of exporting a model for use in the shortfin llm server with an MI300 GPU.
+
+### Pre-Requisites
+- Python >= 3.11 is recommended for this flow
+    - You can check out [pyenv](https://github.com/pyenv/pyenv) as a good tool to be able to manage multiple versions of python on the same system.
+
+### Create Virtual Environment
+To start, create a new virtual environment:
+
+```bash
+python -m venv --prompt shark-ai .venv
+source .venv/bin/activate
+```
+
+### Install `shark-ai`
+You can install either the `latest stable` version of `shark-ai` or the `nightly` version:
+#### Stable
+```bash
+pip install shark-ai
+```
+
+#### Nightly
+```bash
+pip install sharktank -f https://github.com/nod-ai/SHARK-Platform/releases/expanded_assets/dev-wheels
+pip install shortfin -f https://github.com/nod-ai/SHARK-Platform/releases/expanded_assets/dev-wheels
+```
+
+#### Install dataclasses-json
+TODO: This should be included in release:
+```bash
+pip install dataclasses-json
+```
+
+### Define a directory for export files
+Create a new directory for us to export files like `model.mlir`, `model.vmfb`, etc.
+```bash
+mkdir $PWD/export
+export EXPORT_DIR=$PWD/export
+```
+
+### Define Environment Variables
+Define the following environment variables to make running this example a bit easier:
+
+#### Model/Tokenizer Vars
+This example uses the `llama8b_f16.irpa` and `tokenizer.json` files that are pre-existing on the MI300X-3 system.
+You may need to change the paths for your own system.
+```bash
+export MODEL_PARAMS_PATH=/data/llama3.1/8b/llama8b_f16.irpa # Path to existing .irpa file, may need to change w/ system
+export TOKENIZER_PATH=/data/llama3.1/8b/tokenizer.json # Path to existing tokenizer.json, may need to change w/ system
+```
+
+#### Find Available Port
+You can use the following command to find an available port for your server, and save it as the `$PORT` env variable:
+```bash
+export PORT=$(python -c "
+import socket
+
+def find_available_port(starting_port=8000, max_port=8100):
```

Comment:
That's a good point, this is a bit more verbose than what I would typically see in a user doc. This was added as feedback after the initial `user path` walkthrough for the shortfin llm docs. When we did the run through, Kumar started the server at port 8000, but it was already in use. So, we decided that we should have a way for users to check port availability and obtain an available port.

Maybe for this path, we could just have them set the PORT env variable and add a `reminder` that the port has to be free.  Maybe include a cli command that they could run to check availability. Although that may also be a base assumption for anyone using a REST API. 

@kumardeepakamd Any thoughts on this?

Reply:
See what other serving frameworks say:

https://docs.vllm.ai/en/latest/getting_started/quickstart.html#openai-compatible-server

> By default, it starts the server at `http://localhost:8000`. You can specify the address with `--host` and `--port` arguments. 

https://pytorch.org/serve/Troubleshooting.html#deployment-and-config-issues

> “Failed to bind to address: http://127.0.0.1:8080”, port 8080/8081 already in use.
Usually, the port number 8080/8081 is already used by some other application or service, it can be verified by using cmd ss -ntl | grep 8080. There are two ways to troubleshoot this issue either kill the process which is using port 8080/8081 or run Torchserve on different ports other than 8080 & 8081.

Anyone deploying an HTTP server (for ML serving or another application) should already be familiar with hosts and ports. All we really need to say is what the default is and how to change it. We can link to a troubleshooting section like TorchServe's if users are still confused.

Reply:
Removed the port code block, added instructions on how to run the server at a different port, and a brief troubleshooting sentence if the address is already in use